### ISSUE TITLE
Gallery Block margin override specific block align margins

### ISSUE THREAD

**Issue by CreativeDive** [WordPress/gutenberg#17690]
Hey,

the gallery block comes with the default CSS of:

`figure.wp-block-gallery {
    display: block;
    margin: 0;
}`

The `margin: 0` overrides the theme block alignment classes like `.alignleft, .alignright, .alignwide` and `.alignfull`.

For example I'm using the following CSS to define the block alignment classes:

`.alignwide {
		margin-right: calc( 25% - 25vw );
		margin-left: calc( 25% - 25vw );		
	}`

However, the gallery block overrides my custom design styles. I can't see any logic behind why the gallery block has a default margin of "0".

It's not the best practise to write some additional CSS like:

`.alignwide, figure.alignwide {
		margin-right: calc( 25% - 25vw );
		margin-left: calc( 25% - 25vw );		
	}`

Because there are many other cases that need to be adjusted in the CSS as well.

How we can solve this? We could remove the `margin: 0` completely or we change the style target from `figure.wp-block-gallery` to `.wp-block-gallery`.

**Comment by CreativeDive** [comment]
@youknowriad It is also very important to fix this issue for the upcoming 5.3 release :-)

**Comment by CreativeDive** [comment]
At this time with Gutenberg 6.6 the gallery block is unable to get the correct alignment.

```
.aligncenter,
.alignleft,
.alignright,
.alignwide,
.alignfull {
      margin: [CUSTOM MARGIN FROM THEME]
}
```

The upper margin style from a theme has no affect for the gallery block, because the default gallery block style ...

```
figure.wp-block-gallery { 
     display: block; margin: 0; 
}
```
overrides the theme alignment margins completely.

**Comment by CreativeDive** [comment]
@tellthemachines is this already fixed in the latest 6.6 Gutenberg version? The issue is still there after installing 6.6.

**Comment by youknowriad** [comment]
The PR was just merged so it's not on 6.6

**Comment by CreativeDive** [comment]
@youknowriad thanks for the note :-)
